src/main/java/org/example/game2048/board/BoardGenerator.java

```diff
@@ -27,19 +28,22 @@ public static Board updateBoard(Board board) {
         List<Point> points = new ArrayList<>(board.getPointList());
         Point newPoint;
         boolean isNewPointEqual;
-        do {
-            isNewPointEqual = false;
-            newPoint = new Point(RANDOM.nextInt(MAX_SIZE_OF_BOARD), RANDOM.nextInt(MAX_SIZE_OF_BOARD), NEW_VALUE);
-            for (Point x : points) {
-                if (newPoint.equals(x)) {
-                    isNewPointEqual = true;
-                    break;
+        if (points.size() < 16) {
+            do {
+                isNewPointEqual = false;
+                newPoint = new Point(RANDOM.nextInt(MAX_SIZE_OF_BOARD), RANDOM.nextInt(MAX_SIZE_OF_BOARD), NEW_VALUE);
+                for (Point x : points) {
+                    if (newPoint.equals(x)) {
+                        isNewPointEqual = true;
+                        break;
+                    }
                 }
-            }
-        } while (isNewPointEqual);
-        points.add(newPoint);
-        board.setPointList(points);
-        return board;
+            } while (isNewPointEqual);
+            points.add(newPoint);
+            board.setPointList(points);
+            return board;
+        } else return null;
```

Comment:
ten else to z klamrami prosze :) praktycznie nikt w javie nie pisze bez klamer pomimo tego ze to dziala :)